tidy3d/components/heat_charge/devsim_settings.py

```diff
@@ -0,0 +1,63 @@
+"""File containing classes required for the setup of a DEVSIM case."""
+
+from abc import ABC
+from typing import Optional
+
+import pydantic.v1 as pd
+
+from ..base import Tidy3dBaseModel
+from ..types import Union
+
+
+class AbstractDevsimStruct(ABC, Tidy3dBaseModel):
+    """"""
+
+
+class DevsimConvergenceSettings(Tidy3dBaseModel):
+    """This class sets some Devsim parameters.
+
+    Example
+    -------
+    >>> import tidy3d as td
+    >>> devsim_settings = td.DevsimSettings(absTol=1e8, relTol=1e-10, maxIters=30)
+    """
+
+    absTol: Optional[pd.PositiveFloat] = pd.Field(
+        default=1e10,
+        title="Absolute tolerance.",
+        description="Absolute tolerance used as stop criteria when converging towards a solution.",
+    )
+
+    relTol: Optional[pd.PositiveFloat] = pd.Field(
+        default=1e-10,
+        title="Relative tolerance.",
+        description="Relative tolerance used as stop criteria when converging towards a solution.",
+    )
+
+    maxIters: Optional[pd.PositiveInt] = pd.Field(
+        default=30,
+        title="Maximum number of iterations.",
+        description="Indicates the maximum number of iterations to be run. "
+        "The solver will stop either when this maximum of iterations is met "
+        "or when the tolerance criteria has been met.",
+    )
+
+    dV: Optional[pd.PositiveFloat] = pd.Field(
+        1.0,
+        title="Bias step.",
+        description="By default, a solution is computed at 0 bias. "
+        "If a bias different than 0 is requested, DEVSIM will start at 0 and increase bias "
+        "at 'dV' intervals until the required bias is reached. ",
+    )
+
+    num_subprocess: Optional[pd.PositiveInt] = pd.Field(
+        1,
+        title="Number of subprocesses",
+        description="When an array of voltages is provided in 'VoltageBC' it "
+        "is possible to run parallel processes for each of these voltages. This is "
+        "done with the 'multiprocessing' package and 'num_subprocesses "
+        "determines how many solutions to each of the provided voltages is solved at time.",
+    )
```

Comment:
So currently when we set a configuration, I think we do it in: https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/constants.py I guess it's different in this step, but I do wonder if these are instead ("simulation parameters" rather than require a static class definition. Like just thinking if this should be abstracted at the simulation level since it's confiugring a _solver_? Or maybe it makes sense not to do that in this case?

Reply:
I think in this case it makes sense to have them this way. Mostly because these parameters can change from simulation to simulation and you may want to tweak them accordingly. They have default values that may work for some cases but not for sure. For instance, in some cases `dV=5` will be cheaper to run than having the default, and in other cases, you'd need `dV=0.2` for the solution to converge (as is the case in the back-end test).

Reply:
I see. Are these settings something that, say, are defined specifically for DEVSIM only? Or, say when we do implement our own custom solver, I guess some of these parameters will be shared? I am just wondering whether they should be in their own class like this or more-generic parameters of HeatChargeSimulation? Mainly out of curiosity since it sounds these are simulation-specific parameters anyway

Reply:
- Yeah, this class doesn't need to be defined. There'll be default settings in the backend. 
- `ChargeToleranceSpec` sounds good to me too
- `dV` is indeed a steady state. So I think I'll do as you suggest except maybe with a different naming?, i.e.,

```
class: DCSpec:
    dV: pd.PositiveFloat() = ...

class SmallSignalSpec:
    frequencies: Tuple(pd.PositiveFloat()) = ...

class NonlinearTransientSpec:
    dt: pd.PositiveFloat() = ...
    run_time: 
```

Reply:
Nice! `ChargeToleranceSpec` sounds good to me but curious whether HeatChargeSimulation(dV=1, othertoleranceparams...) make more sense? Or would there be any reasons this tolerance spec does not represent a simulation configuration parameter?

Reply:
I agree maybe we need a parameter that has a more comprehensive definition of the goal of dV. For example, eventually we might want to do SPICE integration and I believe we would benefit on following those simulation domains as they are the same: `opearting point`, `transient`, `frequency`, ...  This will also make compiling the equivalent circuit models in the future from the device simulation much easier. So maybe some specification on those lines that can encompass and integrate nicely with SPICE? How would that sound?

Reply:
> > Nice! ChargeToleranceSpec sounds good to me but curious whether HeatChargeSimulation(dV=1, othertoleranceparams...) make more sense? Or would there be any reasons this tolerance spec does not represent a simulation configuration parameter?

> We do this to group settings in contained groups, in the same way we have dl to specify mesh size within a grid spec. Not sure you were referring to this, though

I was also wondering what Dario means here, but yeah, generally we prefer bundling things in Spec-s if they are logically connected.

> What abbreviations? DC? or Spec? Also, We've been using camel case for class names in many places. Are you saying we should be using snake case for those too?

Class names should be camel case. I think he's talking about things like `relTol`, `dV`, `maxIter`